managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2060,7 +2060,7 @@ private void internalReadFromLedger(ReadHandle ledger, OpReadEntry opReadEntry)
             long lastValidEntry = -1L;
             long entryId = firstEntry;
             for (; entryId <= lastEntry; entryId++) {
-                if (opReadEntry.skipCondition.test(PositionImpl.get(ledger.getId(), entryId))) {
+                if (!opReadEntry.skipCondition.test(PositionImpl.get(ledger.getId(), entryId))) {
```

Comment:
I suggest Invert if condition to help simple understand this logic. : )

Reply:
Ok.